apps/ejabberd/src/mod_roster.erl

```diff
@@ -415,15 +441,15 @@ do_process_item_set(JID1,
                 {Item, Item3}
         end,
     case transaction(LServer, F) of
-        {atomic, {OldItem, Item}} ->
-            push_item(User, LServer, To, Item),
-            case Item#roster.subscription of
+        {atomic, {OldItem, NewItem}} ->
+            push_item(User, LServer, To, NewItem),
+            case NewItem#roster.subscription of
                 remove ->
                     send_unsubscribing_presence(From, OldItem), ok;
                 _ -> ok
             end;
         E ->
-            ?DEBUG("ROSTER: roster item set error: ~p~n", [E]), ok
+            ?ERROR_MSG("ROSTER: roster item set error: ~p~n", [E]), ok
```

Comment:
Since we touch log message, could we align to the new log format initiative described in: #1248